workflows/eager.nf

```diff
@@ -292,24 +292,81 @@ workflow EAGER {
     }
 
     //
-    // Section: Metagenomics screening
+    // MODULE: remove reads mapping to the host from the raw fastq
     //
+    if ( params.run_host_removal ) {
+        // Preparing bam channel for host removal to be combined with the input fastq channel
+        // The bam channel consist of [meta, bam, bai] and in the meta we have in addition 'single_end' always set as TRUE and 'reference' set
+        // To be able to join it with fastq channel, we need to remove them from the meta (done in map) and stored in new_meta
+        ch_bam_for_host_removal= MAP.out.bam.join(MAP.out.bai)
+                                            .map{
+                                                meta, bam, bai ->
+                                                new_meta = meta.clone().findAll{ it.key !in [ 'single_end', 'reference' ] }
```

Comment:
We don't do `clone` anymore, as it's not 'safe': https://www.youtube.com/watch?v=A357C-ux6Dw&t=509s&pp=ygURcm9iIHN5bWUgbmV4dGZsb3c%3D

Reply:
This is part of the eager script, I would treat it as separate issue outside the metagenomics PR